### PR TITLE
Fix CLI exit code on entry create/update failure

### PR DESCRIPTION
**Pull request by azdagron**
**Pull Request check list**

- [X] Commit conforms to CONTRIBUTING.md?
- [X] Proper tests/regressions included?
- [X] Documentation updated?

**Affected functionality**
Changes the exit code to non-zero when one or more entries fails to be created or updated using the CLI. It also cleans up some of the error output.

**Description of change**

Recent changes to adopt the batch entry create/update APIs broke the exit code when the operation fails.

This change fixes the exit code.

It also updates the CLI framework to prefix errors with "Error: " to maintain casing consistency with the other messages printed by the CLI.

Also, the CLI output when entry creation/updating fails is a little cryptic.  The status message in particulary is printed after the
"failed" entry following a newline, which causes it to be grouped with the next failed entry (when there is more than one entry that failed). This change updates the output so that the status code and message are grouped clearly with the offending entry. It also prints the text to stderr instead of stdout.

Before the change:
```
FAILED to create the following entries:
Entry ID         :
SPIFFE ID        : spiffe://example.org/workload1
Parent ID        : spiffe://example.org/node2
Revision         : 0
TTL              : default
Selector         : unix:uid:1

similar entry already exists
Entry ID         :
SPIFFE ID        : spiffe://example.org/workload2
Parent ID        : spiffe://example.org/node2
Revision         : 0
TTL              : default
Selector         : unix:uid:2

similar entry already exists
```

After the change:
```
Failed to create the following entry (code:AlreadyExists, msg:"similar entry already exists"):
Entry ID         : (none)
SPIFFE ID        : spiffe://example.org/workload1
Parent ID        : spiffe://example.org/node2
Revision         : 0
TTL              : default
Selector         : unix:uid:1

Failed to create the following entry (code:AlreadyExists, msg:"similar entry already exists"):
Entry ID         : (none)
SPIFFE ID        : spiffe://example.org/workload2
Parent ID        : spiffe://example.org/node2
Revision         : 0
TTL              : default
Selector         : unix:uid:2

Error: failed to create one or more entries
```

**Which issue this PR fixes**
Fixes: #1977

### CODE REVIEW

File: cmd/spire-server/cli/entry/create.go [hunk 1]
```diff
@@ -106,16 +106,19 @@ func (c *createCommand) Run(ctx context.Context, env *common_cli.Env, serverClie
...
 	}
 
 	// Print entries that failed to be created
-	if len(failed) > 0 {
-		env.Printf("FAILED to create the following %s:\n", util.Pluralizer("", "entry", "entries", len(failed)))
-	}
 	for _, r := range failed {
-		printEntry(r.Entry, env)
-		env.Printf("%s\n", r.Status.Message)
+		env.ErrPrintf("Failed to create the following entry (code:%s, msg:%q):\n",
```
Comment (amartinezfayo) [inline cmd/spire-server/cli/entry/create.go:114]:
```suggestion
		env.ErrPrintf("Failed to create the following entry (code: %s, msg: %q):\n",
```

File: cmd/spire-server/cli/entry/update.go [hunk 2]
```diff
@@ -106,16 +106,19 @@ func (c *updateCommand) Run(ctx context.Context, env *common_cli.Env, serverClie
...
 	}
 
 	// Print entries that failed to be updated
-	if len(failed) > 0 {
-		env.Printf("FAILED to update the following %s:\n", util.Pluralizer("", "entry", "entries", len(failed)))
-	}
 	for _, r := range failed {
-		printEntry(r.Entry, env)
-		env.Printf("%s\n", r.Status.Message)
+		env.ErrPrintf("Failed to update the following entry (code:%s, msg:%q):\n",
```
Comment (amartinezfayo) [inline cmd/spire-server/cli/entry/update.go:114]:
```suggestion
		env.ErrPrintf("Failed to update the following entry (code: %s, msg: %q):\n",
```

**Review by amartinezfayo — COMMENTED** [review]
Thanks for the fix @azdagron!
Only a very minor suggestion before merging.

File: cmd/spire-server/cli/entry/create.go (hunk 1 above)
Comment (azdagron) [reply cmd/spire-server/cli/entry/create.go]: :+1:

File: cmd/spire-server/cli/entry/update.go (hunk 2 above)
Comment (azdagron) [reply cmd/spire-server/cli/entry/update.go]: :+1:

**Review by amartinezfayo — APPROVED** [review]

**Pull request merged**